### DISCUSSION TITLE
Code style

### DISCUSSION THREAD

**Discussion by dluc** [awaescher/OllamaSharp#81] [General]
I checked out the repo and noticed that indentation uses tabs, and async methods are missing the usual "Async" suffix. These lead to code style error [IDE0055](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0055) and warning [VSTHRD200](https://github.com/microsoft/vs-threading/blob/main/doc/analyzers/VSTHRD200.md).

While these can be ignored or disabled, I was curious if this was an intentional choice.

The tab usage can be an inconvenience for contributors whose IDEs are set to use spaces, though there are tools to handle this automatically.

However, the absence of the `Async` suffix (e.g. in `Chat.SendAs`) is more noticeable for users of the library, potentially causing confusion when determining whether to use `await` with method calls.

Would you consider addressing these in the next major release?

**Comment by awaescher** [comment]
Yes, as stated in the .editorconfig, tabs are indentation of choice.

Also, the Async suffix is deviding the developers. I don't think it's useful except there are sync overloads of a method too. There is no combination of sync and async methods in OllamaSharp, that's why I kept it clean without the Async suffix.

**Reply by JerrettDavis** [reply]
@awaescher For version 4, should we consider supplying both async and synchronous methods?

**Reply by awaescher** [reply]
I do see a necessity, do you?

**Reply by JerrettDavis** [reply]
I don't believe including synchronous methods is necessary, and it may not be the best direction for the library. I do, however, prefer sticking to the `Async` naming convention, as it helps avoid confusion about the nature of the methods. That said, with the library having no synchronous calls today, I don't think this change is necessary either. Adding synchronous methods could inadvertently encourage developers to use the library in unintended ways, particularly since the operations we handle are inherently asynchronous. 

Additionally, introducing synchronous methods would create extra maintenance overhead for what I suspect would be rarely, if ever, used functionality.

Ultimately, my goal in raising this is to help us establish clear coding guidelines that foster a consistent and intuitive codebase for our users.

**Comment by awaescher** [comment]
I have been in touch with Microsoft regarding #112 which will lead to breaking changes and a version bump to 4.0.
With that opportunity, I will append the Async suffix, but won't add synchronous methods.
